lua/oil/util.lua

```diff
@@ -858,4 +860,30 @@ M.get_edit_path = function(bufnr, entry, callback)
   end
 end
 
+--- Check for an icon provider and return a common icon provider API
+---@return (oil.IconProvider)?
+M.get_icon_provider = function()
+  -- prefer mini.icons
+  local has_mini_icons, mini_icons = pcall(require, "mini.icons")
```

Comment:
@mehalter, this is not the recommended way for plugins to determine if 'mini.icons' should be used. See [here](https://github.com/echasnovski/mini.nvim/blob/d187614ecee84ef8d7456339ed7ce1160d054310/doc/mini-icons.txt#L248-L250). The reason is that `pcall(require, 'mini.icons')` would return `true` if user has the whole 'mini.nvim' installed, which would conflict with possible idea of using 'nvim-web-devicons' instead of 'mini.icons'.

Reply:
Ah thanks! I will also check for `_G.MiniIcons` with a follow up. We definitely want to do the `require` at least to protect against the user doing any sort of lazy loading.

Reply:
Hm I suppose if the user _isn't_ using `mini.icons` but other modules in the entire `mini.nvim` module then this would _force_ the setup 🤔 I suppose we should just check for the global function and hope the user doesn't lazy load it.

Reply:
> We definitely want to do the `require` at least to protect against the user doing any sort of lazy loading.

> We can leave the `require` and protect users from lazy loading 🎉

I am not really sure what you mean here. My main point is to make sure that the approach here works for users which install whole 'mini.nvim' instead of standalone 'mini.icons' (which is the recommended way) but still want to use 'nvim-web-devicons'.

Reply:
Yeah I totally understand your main point 😄 I just had a few minutes to open up a PR to resolve the issue: https://github.com/stevearc/oil.nvim/pull/441

Thanks for pointing this out!